include/linux/soundwire/sdw.h

```diff
@@ -561,6 +565,8 @@ struct sdw_slave {
 	struct list_head node;
 	struct completion *port_ready;
 	u16 dev_num;
+	bool probed;
+	struct completion probe_complete;
```

Comment:
@plbossart  I checked the commit message carefully, and I wanted to give a more detailed and concrete example about following segment.
````
The first race condition which can happen is between the .probe(),
which allocates the resources, and .update_status() where
initializations are typically handled. The .probe() is only called
once during the initial boot, while .update_status() will be called
for every bus hardware reset and if the Slave device loses
synchronization (an unlikely event but with non-zero probability).

The time difference between the end of the enumeration process and the
call to .update_status() is as small as a SoundWire frame, so with a
48kHz frame rate, .update_status() may be invoked in 10s of
microseconds.

Since there is no guarantee that the .probe() callback completes by
the time .update_status() is invoked, it's not unusual for the
.update_status() to rely on data structures that have not been
allocated yet, leading to kernel oopses.
````

.update_status() will be called in a interrupt thread for every bus hardware reset and if the Slave device loses
synchronization. When slave is being probed, it is very likely be preempted by a interrupt triggered by attach event from slave, then in interrupt thread, .update_status() will try to update hw which is still not initialized by .probe().

Reply:
ok, will reword. thanks!